### PR TITLE
Create system banner API handler

### LINKED ISSUE #2603: Provide a way for cloud providers to show a system banner
Feature proposal:
Allow cloud providers to configure Dashboard UI to show a system banner. The banner would be configurable via command line flags. Admins would be able to set its prominence and style. 

Use cases for this:
1. Showing cloud provider specific banner, e.g., "To edit this cluster go [there]". 
2. Showing system messages, e.g., "This cluster is going into maintenance mode next week [learn more]".

@maciaszczykm @floreks @danielromlein What do you think of contributing this?

### PR DESCRIPTION
**Pull request by maciaszczykm**
Closes #2603.

### CODE REVIEW

**Comment by bryk** [comment]
I'll review tomorrow.

**Comment by maciaszczykm** [comment]
@bryk It's a bit too early for a review. It would be better for you to wait for frontend part :)

**Comment by bryk** [comment]
:)

Backend code looks good. For the frontend, how about we put it in an action bar-like thing on the top. Or below the blue part. 

I think I prefer below the blue part. Your opinions?

**Comment by maciaszczykm** [comment]
@bryk I agree.

File: src/app/backend/systembanner/api/types.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+// SystemBanner represents system banner.
+type SystemBanner struct {
+	Message  string `json:"message"`
+	Severity string `json:"severity"`
+}
+
+// GetSeverity returns one of allowed severity values based on given parameter.
+func GetSeverity(severity string) string {
```
Comment (floreks) [inline src/app/backend/systembanner/api/types.go]: I'd add type that maps to a string (i.e. `type BannerSeverity string`) so we can return our type here. It's easier to find out correct values later on.

Maybe also use a switch on string here as it feels like a perfect place for it.

File: src/app/backend/systembanner/api/types.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+	// SystemBannerSeverityInfo is in the middle of allowed system banner severities.
+	SystemBannerSeverityWarning = "WARNING"
+
+	// SystemBannerSeverityInfo is the highest of allowed system banner severities.
+	SystemBannerSeverityError = "ERROR"
+)
+
+// SettingsManager is used for user settings management.
+type SettingsManager interface {
```
Comment (floreks) [inline src/app/backend/systembanner/api/types.go]: SettingsManager?

File: src/app/backend/systembanner/api/types.go (hunk 1 above)
Comment (bryk) [reply src/app/backend/systembanner/api/types.go]: +1

Comment (maciaszczykm) [reply src/app/backend/systembanner/api/types.go]: Done.

File: src/app/backend/systembanner/api/types.go (hunk 2 above)
Comment (maciaszczykm) [reply src/app/backend/systembanner/api/types.go]: Done.

**Comment by bryk** [comment]
Ping me when this is ready for rereview.

**Comment by maciaszczykm** [comment]
image

image

**Comment by maciaszczykm** [comment]
@bryk PTAL

**Comment by bryk** [comment]
I think we should make this banner 48px. It looks alike other toolbars. 24px seemed too small to me. WDYT? 

selection_001

File: src/app/frontend/chrome/component.js [hunk 3]
```diff
@@ -50,11 +52,60 @@ export class ChromeController {
...
+  getSystemBannerClass() {
+    if (this.systemBanner_ && this.systemBanner_.severity) {
+      return `kd-system-banner-${this.systemBanner_.severity.toLowerCase()}`;
+    }
+    return '';
+  }
+
+  /**
+   * @export
+   * @return {*}
```
Comment (bryk) [inline src/app/frontend/chrome/component.js]: {sting}? If possible

Comment (floreks) [reply src/app/frontend/chrome/component.js]: Unfortunately not. `trustAsHtml` does not have precise return type and it can be `{Object|string|null}`

Comment (maciaszczykm) [reply src/app/frontend/chrome/component.js]: Unfortunately not possible.

**Comment by maciaszczykm** [comment]
zrzut ekranu 2017-11-24 o 11 17 26

**Pull request merged**